File: Human_written_code/Code_with_comments/CPP/Q31.cpp
#include <iostream>
using namespace std;

int main() {
    int t;
    cin >> t;  // Read the number of test cases

    while (t--) {
        int a, b;
        cin >> a >> b;  // Read two integers

        // Check if the sum of a and b is even or odd
        if ((a + b) % 2 == 0)
            cout << "Bob" << endl;  // Output "Bob" if sum is even
        else
            cout << "Alice" << endl;  // Output "Alice" if sum is odd
    }

    return 0;
}

File: Human_written_code/Code_with_comments/Python/Q31.py
def main():
    # Read the number of test cases
    t = int(input().strip())
    
    # Loop through each test case
    for _ in range(t):
        # Read two integers a and b
        a, b = map(int, input().split())
        
        # Check if the sum of a and b is even
        if (a + b) % 2 == 0:
            print("Bob")
        else:
            print("Alice")

if __name__ == "__main__":
    main()
